src/sentry/auth/helper.py

```diff
@@ -457,6 +467,21 @@ def handle_unknown_identity(
             op = None
 
         if not op:
+            if self.user.is_authenticated:
+                template = "auth-confirm-link"
+                existing_user = self.user
+            else:
+                existing_user = acting_user or self._get_user(identity)
+                if existing_user and features.has(
+                    "organizations:idp-automatic-migration", self.organization
+                ):
+                    create_verification_key(existing_user, self.organization, identity["email"])
+                    template = "auth-confirm-account"
+                else:
+                    self.request.session.set_test_cookie()
+                    template = "auth-confirm-identity"
+                    existing_user = acting_user
```

Comment:
this confused me a little bit along with the above, i'm guessing that you wanted to get rid of the `context.update`, but that was more clear to me than redefining a variable.

Reply:
maybe it would be most clear if existing_user was first declared as None, and is either set in the first if block to `self._get_user_idenity`, or in this block to `acting_user`

Reply:
Yes, factoring out the `context.update` was my main motivation. Redefining the `existing_user` variable was sort of opportunistic... it was a natural name for the value being newly queried for, and when I noticed that `"existing_user"` was the existing `context` label it was like, hey, maybe it's actually easier to understand this way?

I think I can make it easier to understand along with the other changes, in the extracted helper method that I mentioned above.